saw-core-what4/src/Verifier/SAW/Simulator/What4.hs

```diff
@@ -1431,7 +1431,7 @@ reconstructArgTerm atrm sc ts =
              pure (x, ts1)
         ArgTermVector ty ats ->
           do (xs, ts1) <- parseList ats ts0
-             x <- scVector sc ty xs
+             x <- scVectorReduced sc ty xs
```

Comment:
What motivated this change? It's not clear to me how this relates to the tactic changes elsewhere in the patch.

Reply:
this it not related, I forgot it was in the branch. This simplifies the output of `goal_eval` such that `[x@0, x@1]` becomes `x`, when `x` has exactly 2 elements. Do you want me to split it in a different pull request?

Reply:
If you have another PR in the works, then yes, I think it would be better suited for that. If not, don't worry about it too much.